File: app/src/main/java/de/bahnhoefe/deutschlands/bahnhofsfotos/InboxActivity.java
package de.bahnhoefe.deutschlands.bahnhofsfotos;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import java.util.List;

import de.bahnhoefe.deutschlands.bahnhofsfotos.databinding.ActivityInboxBinding;
import de.bahnhoefe.deutschlands.bahnhofsfotos.db.InboxAdapter;
import de.bahnhoefe.deutschlands.bahnhofsfotos.model.PublicInbox;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class InboxActivity extends AppCompatActivity {

    private static final String TAG = InboxActivity.class.getSimpleName();

    private InboxAdapter adapter;

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        final var binding = ActivityInboxBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        final Call<List<PublicInbox>> inboxCall = ((BaseApplication)getApplication()).getRsapiClient().getPublicInbox();
        inboxCall.enqueue(new Callback<>() {
            @Override
            public void onResponse(@NonNull final Call<List<PublicInbox>> call, @NonNull final Response<List<PublicInbox>> response) {
                final var body = response.body();
                if (response.isSuccessful() && body != null) {
                    adapter = new InboxAdapter(InboxActivity.this, body);
                    binding.inboxList.setAdapter(adapter);
                    binding.inboxList.setOnItemClickListener((parent, view, position, id) -> {
                        final var inboxItem = body.get(position);
                        final var intent = new Intent(InboxActivity.this, MapsActivity.class);
                        intent.putExtra(MapsActivity.EXTRAS_LATITUDE, inboxItem.getLat());
                        intent.putExtra(MapsActivity.EXTRAS_LONGITUDE, inboxItem.getLon());
                        intent.putExtra(MapsActivity.EXTRAS_MARKER, inboxItem.getStationId() == null ? R.drawable.marker_missing : R.drawable.marker_red);
                        startActivity(intent);
                    });
                }
            }

            @Override
            public void onFailure(@NonNull final Call<List<PublicInbox>> call, @NonNull final Throwable t) {
                Log.e(TAG, "Error loading public inbox", t);
                Toast.makeText(getBaseContext(), getString(R.string.error_loading_inbox) + t.getMessage(), Toast.LENGTH_LONG).show();
            }
        });
    }

}

File: app/src/main/kotlin/de/bahnhoefe/deutschlands/bahnhofsfotos/InboxActivity.kt
package de.bahnhoefe.deutschlands.bahnhofsfotos

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.ViewGroup.MarginLayoutParams
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.updateLayoutParams
import dagger.hilt.android.AndroidEntryPoint
import de.bahnhoefe.deutschlands.bahnhofsfotos.databinding.ActivityInboxBinding
import de.bahnhoefe.deutschlands.bahnhofsfotos.db.InboxAdapter
import de.bahnhoefe.deutschlands.bahnhofsfotos.model.PublicInbox
import de.bahnhoefe.deutschlands.bahnhofsfotos.rsapi.RSAPIClient
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import javax.inject.Inject

private val TAG = InboxActivity::class.java.simpleName

@AndroidEntryPoint
class InboxActivity : AppCompatActivity() {
    private var adapter: InboxAdapter? = null

    @Inject
    lateinit var rsapiClient: RSAPIClient

    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge()
        super.onCreate(savedInstanceState)
        val binding = ActivityInboxBinding.inflate(layoutInflater)
        ViewCompat.setOnApplyWindowInsetsListener(binding.getRoot()) { v, windowInsets ->
            val insets = windowInsets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.updateLayoutParams<MarginLayoutParams> {
                bottomMargin = insets.bottom
                topMargin = insets.top
            }
            WindowInsetsCompat.CONSUMED
        }
        setContentView(binding.root)
        val inboxCall = rsapiClient.getPublicInbox()
        inboxCall.enqueue(object : Callback<List<PublicInbox>> {
            override fun onResponse(
                call: Call<List<PublicInbox>>,
                response: Response<List<PublicInbox>>
            ) {
                val body = response.body()
                if (response.isSuccessful && body != null) {
                    adapter = InboxAdapter(this@InboxActivity, body)
                    binding.inboxList.adapter = adapter
                    binding.inboxList.onItemClickListener =
                        OnItemClickListener { _: AdapterView<*>?, _: View?, position: Int, _: Long ->
                            val (_, _, stationId, lat, lon) = body[position]
                            val intent = Intent(this@InboxActivity, MapsActivity::class.java)
                            intent.putExtra(EXTRAS_MAPS_LATITUDE, lat)
                            intent.putExtra(EXTRAS_MAPS_LONGITUDE, lon)
                            intent.putExtra(
                                EXTRAS_MAPS_MARKER,
                                if (stationId == null) R.drawable.marker_missing else R.drawable.marker_red
                            )
                            startActivity(intent)
                        }
                }
            }

            override fun onFailure(call: Call<List<PublicInbox>>, t: Throwable) {
                Log.e(TAG, "Error loading public inbox", t)
                Toast.makeText(
                    baseContext,
                    getString(R.string.error_loading_inbox) + t.message,
                    Toast.LENGTH_LONG
                ).show()
            }
        })
    }

}